.github/workflows/yocto-build-deploy.yml

```
@@ -673,8 +680,8 @@ jobs:
           RELEASE_VERSION: "${{ steps.balena-lib.outputs.os_version }}"
           BOOTABLE: 1
           TRANSLATION: "v6"
-          FINAL: ${{ steps.merge-test-result.outputs.finalize || inputs.finalize-hostapp }}
-          ESR: "${{ inputs.deploy-esr }}"
+          FINAL: ${{ steps.should-finalize.outputs.finalize }}
+          ESR: "${{ steps.esr-check.outputs.is-esr }}"
```

```suggestion
          ESR: "${{ steps.esr-check.outcome == 'success' }}"
```
Here is another way to write this without relying on outputs. Up to you which you prefer.